File: 5th_semester/functional_programming/haskell/task2.hs
data Tree = Empty | Node Int Tree Tree deriving (Show, Eq)
mulTrees :: Tree -> Tree -> Tree
mulTrees Empty Empty = Empty
mulTrees (Node a b c) Empty = Node a b c
mulTrees Empty (Node a b c) = Node a b c
mulTrees (Node a1 b1 c1) (Node a2 b2 c2) = Node (a1 * a2) (mulTrees b1 b2) (mulTrees c1 c2)

main = print(mulTrees (Node 18 Empty Empty) (Node 5 (Node 3 Empty Empty) Empty))

File: 7th_semester/prolog/test/task2.pl
numLabels(empty, _, 0). 
numLabels(tree(V, L, R), V, N) :- !, numLabels(L, V, N1), 
numLabels(R, V, N2), 
N is N1 + N2 + 1. 
numLabels(tree(_, L, R), V1, N) :- numLabels(L, V1, N1), 
numLabels(R, V1, N2), 
N is N1 + N2.
